src/server/pfs/cmds/cmds.go

```diff
@@ -1071,47 +1079,105 @@ $ {{alias}} foo@master:path1 bar@master:path2`,
 				return err
 			}
 			defer c.Close()
 
-			newFiles, oldFiles, err := c.DiffFile(
-				newFile.Commit.Repo.Name, newFile.Commit.ID, newFile.Path,
-				oldFile.Commit.Repo.Name, oldFile.Commit.ID, oldFile.Path,
-				shallow,
-			)
-			if err != nil {
-				return err
-			}
-
-			if len(newFiles) > 0 {
-				fmt.Println("New Files:")
-				writer := tabwriter.NewWriter(os.Stdout, pretty.FileHeader)
-				for _, fileInfo := range newFiles {
-					pretty.PrintFileInfo(writer, fileInfo, fullTimestamps, false)
+			return pager.Page(noPager, os.Stdout, func(w io.Writer) (retErr error) {
+				var writer *tabwriter.Writer
+				if nameOnly {
+					writer = tabwriter.NewWriter(w, pretty.DiffFileHeader)
+					defer func() {
+						if err := writer.Flush(); err != nil && retErr == nil {
+							retErr = err
+						}
+					}()
 				}
-				if err := writer.Flush(); err != nil {
+
+				newFiles, oldFiles, err := c.DiffFile(
+					newFile.Commit.Repo.Name, newFile.Commit.ID, newFile.Path,
+					oldFile.Commit.Repo.Name, oldFile.Commit.ID, oldFile.Path,
+					shallow,
+				)
+				if err != nil {
 					return err
 				}
-			}
-			if len(oldFiles) > 0 {
-				fmt.Println("Old Files:")
-				writer := tabwriter.NewWriter(os.Stdout, pretty.FileHeader)
-				for _, fileInfo := range oldFiles {
-					pretty.PrintFileInfo(writer, fileInfo, fullTimestamps, false)
-				}
-				if err := writer.Flush(); err != nil {
-					return err
+
+				nI, oI := 0, 0
+				for {
+					if nI == len(newFiles) && oI == len(oldFiles) {
+						break
+					}
+					if err := func() (retErr error) {
+						var nFI *pfsclient.FileInfo
+						var oFI *pfsclient.FileInfo
+						switch {
+						case oI == len(oldFiles) || newFiles[nI].File.Path < oldFiles[oI].File.Path:
+							nFI = newFiles[nI]
+							nI++
+						case nI == len(newFiles) || oldFiles[oI].File.Path < newFiles[nI].File.Path:
+							oFI = oldFiles[oI]
+							oI++
+						case newFiles[nI].File.Path == oldFiles[oI].File.Path:
+							nFI = newFiles[nI]
+							nI++
+							oFI = oldFiles[oI]
+							oI++
+						}
+						nPath, oPath := "/dev/null", "/dev/null"
+						if nFI != nil {
+							if nameOnly {
+								pretty.PrintDiffFileInfo(writer, true, nFI, fullTimestamps)
+							} else {
+								nPath, err = dlFile(c, nFI.File)
+								if err != nil {
+									return err
+								}
+								defer func() {
+									if err := os.RemoveAll(nPath); err != nil && retErr == nil {
+										retErr = err
+									}
+								}()
+							}
+						}
+						if oFI != nil {
+							if nameOnly {
+								pretty.PrintDiffFileInfo(writer, false, oFI, fullTimestamps)
+							} else {
+								oPath, err = dlFile(c, oFI.File)
+								defer func() {
+									if err := os.RemoveAll(oPath); err != nil && retErr == nil {
+										retErr = err
+									}
+								}()
+							}
+						}
+						if nameOnly {
+							return nil
+						}
+						cmd := exec.Command("git", "-c", "color.ui=always", "--no-pager", "diff", "--no-index", oPath, nPath)
```

Comment:
It does not, I'm thinking we should degrade to `diff` if `git` isn't present, but I didn't want to use `diff` by default because `git` is so much nicer and its diff output is so much better known. I'll also make the command configurable.